Coding style updates including async/await, var, and Code Cleanup

mikebattista opened the pull request.

mikebattista committed:
Ran Code Cleanup.

mikebattista committed:
Use var throughout package functions for consistency.

mikebattista committed:
Use async/await.

mikebattista committed:
Removed whitespace.

ptorr-msft commented on Samples/PackageWithExternalLocation/cs/PhotoStoreDemo/ExecutionMode.cs:
```diff
@@ -1,9 +1,6 @@
-using System.Threading.Tasks;
 
 namespace PhotoStoreDemo
 {
```
Line 25 - `result != 15700` (which I assume is `APPMODEL_ERROR_NO_PACKAGE`) is very obscure; the API returns `ERROR_SUCCESS` when successful, which is much clearer, so just look for `return == 0`.

ptorr-msft replied:
Also you don't need the "Is Windows 7" test - if it's not WIn7 then I am pretty sure you will crash in the JIT when it can't find `GetCurrentPackageFullName` - you need to double-wrap the function and check for `MissingMethodException` and other gymnastics. I would just ignore Win7 since it's out of support now.

mikebattista replied:
Fixed and removed.

ptorr-msft commented on Samples/PackageWithExternalLocation/cs/PhotoStoreDemo/ExecutionMode.cs:
```diff
@@ -42,7 +39,7 @@ internal static string GetSafeAppxLocalFolder()
             try
             {
-               // return Windows.Storage.ApplicationData.Current.LocalFolder.Path;
+                // return Windows.Storage.ApplicationData.Current.LocalFolder.Path;
```
This method is useless... we should either just delete it or uncomment out the useful line :)

mikebattista replied:
Removed.

ptorr-msft commented on Samples/PackageWithExternalLocation/cs/PhotoStoreDemo/ImageFile.cs:
```diff
@@ -40,26 +39,26 @@ public ImageFile(Stream bitMapStream)
 
         }
 
         public void AddToCache(Stream bitMapStream)
```
Also a fully-commented-out function?

mikebattista replied:
Removed.

ptorr-msft reviewed (commented):
I realize there is too much here for me to review (it's a lot of functional stuff about image editing) but we should kill the Win7 tests and all the empty functions.

mikebattista committed:
Removed unneeded code.

mikebattista committed:
Removed commented code.

mikebattista commented:
> I realize there is too much here for me to review (it's a lot of functional stuff about image editing) but we should kill the Win7 tests and all the empty functions.

I addressed your comments. Let me know if there's anything else.

The main priority is making sure the package registration code is copy/pastable and follows standard calling conventions.
